test/cluster-setup.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+# Copyright 2018 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script calls out to scripts in knative/test-infra to setup a cluster
+# and deploy the Pipeline CRD to it for running integration tests.
+
+source $(dirname $0)/../vendor/github.com/knative/test-infra/scripts/e2e-tests.sh
+
+set -o xtrace
+set -o errexit
+set -o pipefail
+
+function take_down_pipeline() {
+    header "Tearing down Pipeline CRD"
+    ko delete --ignore-not-found=true -f config/
+}
+trap take_down_pipeline EXIT
```

Comment:
unfortunately when running locally, the helper script will invoke `teardown` _before_ running anything, which against a pre-existing cluster caused the subsequent deployment to sometimes fail because during the deployment, we would try to deploy to namespaces that were in the process of being terminated

(ill add a comment!)

Reply:
Yes, when running locally `teardown` is called to ensure a clean state. You could use some existing checks in teardown to avoid the situation you described, like `wait_until_object_does_not_exist`.